### PR TITLE
Add API tests for artist/id/subscribe using stripe-mock

### PR DESCRIPTION
**Pull request by ksa618**
I had to change from `res.status(...)` to `ApiError` for errors. The tests would just hang, waiting for the call to return before I did the change.

### CODE REVIEW

File: src/utils/stripe.ts [hunk 1]
```diff
@@ -18,12 +18,16 @@ import {
 } from "./handleFinishedTransactions";
 import countryCodesCurrencies from "./country-codes-currencies";
 
-const { STRIPE_KEY, API_DOMAIN } = process.env;
+// STRIPE_HOST, STRIPE_PORT, and STRIPE_PROTOCOL should not be set when not running tests against stripe-mock
+const { STRIPE_KEY, API_DOMAIN, STRIPE_HOST, STRIPE_PORT, STRIPE_PROTOCOL } = process.env;
```
Comment (simonv3) [inline src/utils/stripe.ts:22]: should these be included in .env.example as comments? or can we do an if statement that's something if ENV === 'test' or CI = true?

**Review by simonv3 — COMMENTED** [review]
lgtm but do have that one question

File: src/utils/stripe.ts (hunk 1 above)
Comment (ksa618) [reply src/utils/stripe.ts:22]: I don't think they should be in .env.example. The stripe-mock is only for automatic tests, so it wouldn't make sense to set them when running the api containers for manual testing. It's better to use stripe test mode there. Having an if statement is probably a good idea. I'll change it to only use those env variables in test mode.

Comment (ksa618) [reply src/utils/stripe.ts:22]: I tried it and it works, but I have to set `NODE_ENV: test` for `api-test` in the docker-compose.test.yml file (it's currently "development"). The tests pass when I do this, but I see there are a few checks for `NODE_ENV === "development"` in the code. Is it a problem that it's "test" when the tests run?

Comment (simonv3) [reply src/utils/stripe.ts:22]: I'm looking at the code and I think if the tests pass I think it's okay. The node_env === 'dev stuff looks like it's mainly to make some things "more permissive" like to not have rate limiting, for cors (not an issue in the back-end tests). So I think it's okay?

**Review by simonv3 — APPROVED** [review]

File: src/utils/stripe.ts (hunk 1 above)
Comment (simonv3) [reply src/utils/stripe.ts:22]: does the stripe image load these into the env environment? I guess I'm not clear on where the environment variables are coming from

Comment (ksa618) [reply src/utils/stripe.ts]: I've set the in docker-compose.test.yml.

**Review by simonv3 — APPROVED** [review]

File: src/utils/stripe.ts (hunk 1 above)
Comment (simonv3) [reply src/utils/stripe.ts]: Amazing! feel free to merge it when you're happy with it!

**Pull request merged**